### PR TITLE
Update README and add doxygen content

### PR DESCRIPTION
**Pull request by aggarw13**
This PR makes documentation updates to README.md and adds content for Overview page of doxygen output. Besdies, this PR also makes fixes for doxygen output from source code generated for data types and constants.

### CODE REVIEW

File: docs/doxygen/pages.dox [hunk 1]
```diff
@@ -12,10 +21,18 @@
...
 @section sntp_design Design
-@brief SNTP Library Design
+@brief <b>coreSNTP Library Design</b>
 
-<h3>Memory Usage</h3>
 <p>
+
+The coreSNTP library provides 2 layers of API:
+ 1. <b>Serializer/Deserializer and Utilities</b> - This layer provides functionality of serializing SNTP time requests, de-serializing SNTP response packets, and some utility functions helpful in setting up an SNTP client in application.
+ 2. <b>Managed Client</b> - This layer provides **managed** functionality of network operations for DNS resolution, sending and receiving SNTP packets over UDP, authenticating server for security (if enabled) and handling of
```
Comment (archigup) [inline docs/doxygen/pages.dox:30]:
```suggestion
 2. <b>Managed Client</b> - This layer provides **managed** functionality for network operations including DNS resolution, sending and receiving SNTP packets over UDP, authenticating servers for security (if enabled), and handling of
```

File: docs/doxygen/pages.dox [hunk 2]
```diff
@@ -61,9 +78,22 @@
...
 /**
-@defgroup json_enum_types Enumerated Types
+@defgroup sntp_enum_types Enumerated Types
 @brief Enumerated types of the coreSNTP library
+*/
 
-@defgroup json_struct_types Struct Types
+/**
+@defgroup sntp_callback_types Callback Types
+@brief Callback function pointer types of the MQTT library
```
Comment (archigup) [inline docs/doxygen/pages.dox]:
```suggestion
@brief Callback function pointer types of the SNTP library
```

File: docs/doxygen/pages.dox [hunk 3]
```diff
@@ -1,9 +1,18 @@
 /**
 @mainpage Overview
 @anchor sntp
-@brief coreSNTP Library
+@brief <h2><b>coreSNTP Library</b></h2>
```
[the remark this answers is not in the document]
Comment (aggarw13) [reply docs/doxygen/pages.dox:4]: Will remove it

File: docs/doxygen/pages.dox [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
 
+The coreSNTP library provides a client for <b>Simple Time Network Protocol (SNTP)</b> to allow devices to synchronize their system clocks with internet time. This library implements the SNTPv4 specification defined in [RFC 4330](https://tools.ietf.org/html/rfc4330).
 <p>
+An SNTP client can request time from both NTP and SNTP servers. According to the SNTPv4 specification, <i>To an NTP or SNTP server, NTP and SNTP clients are indistinguishable; to an NTP or SNTP client, NTP and SNTP servers are indistinguishable.</i>, thereby, allowing SNTP clients to request time from NTP servers.
```
[the remark this answers is not in the document]
Comment (aggarw13) [reply docs/doxygen/pages.dox]: My intention here is to mention the exact sentence from the RFC. I have updated the formatting for it to look like:
image

**Review by archigup — APPROVED** [review]

**Review by cobusve — APPROVED** [review]

**Pull request merged**